src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -351,6 +367,13 @@ void evaluate_impl(
       "Cannot assign a tensor expression to a LHS tensor with generic "
       "indices that would be contracted, e.g. evaluate<ti::A, ti::a>.");
 
+  if constexpr (tenex::detail::is_vector<X>::value) {
+    ASSERT(get_size((*lhs_tensor)[0]) > 0,
+           "Tensors with vector components must be sized before calling "
+           "tenex::evaluate<...>("
+           "\tgsl::not_null<Tensor<VectorType, ...>*>, number).");
```

Comment:
I put it here in the hopes that it would make the function more readable in an error message, but looking at this I realize I forgot to add newlines in front of the line above it and in front of the tab, so this would look weird as is.

I'll add newlines unless you think I should scrap the newlines and tabs altogether

Reply:
That makes sense.  Tabs and newlines sound fine.